docs/examples/tls_mutual_authentication.rst

```diff
@@ -1,60 +1,44 @@
 
+See https://www.rabbitmq.com/ssl.html for certificate generation and RabbitMQ TLS configuration.
 
 tls_example.py::
 
-    import ssl
-    import pika
     import logging
+    import pika
+    import ssl
 
     logging.basicConfig(level=logging.INFO)
-
-    context = ssl.SSLContext(ssl.PROTOCOL_TLSv1)
-    context.verify_mode = ssl.CERT_REQUIRED
-    context.load_verify_locations('/Users/me/tls-gen/basic/testca/cacert.pem')
-    context.load_cert_chain('/Users/me/tls-gen/basic/client/cert.pem',
-                            '/Users/me/tls-gen/basic/client/key.pem')
-
-    cp = pika.ConnectionParameters(ssl_options=pika.SSLOptions(context))
-
-    conn = pika.BlockingConnection(cp)
-    ch = conn.channel()
-    print(ch.queue_declare("sslq"))
-    ch.publish("", "sslq", "abc")
-    print(ch.basic_get("sslq"))
-
+    context = ssl.create_default_context(
+        cafile="PIKA_DIR/testdata/certs/ca_certificate.pem")
+    context.load_cert_chain("PIKA_DIR/testdata/certs/client_certificate.pem",
+                            "PIKA_DIR/testdata/certs/client_key.pem")
+    ssl_options = pika.SSLOptions(context, "localhost")
+    conn_params = pika.ConnectionParameters(port=5671,
+                                            ssl_options=ssl_options)
+    
+    with pika.BlockingConnection(conn_params) as conn:
+        ch = conn.channel()
+        ch.queue_declare("foobar")
+        ch.publish("", "foobar", "Hello, world!")
+        print(ch.basic_get("foobar"))
 
 rabbitmq.config::
 
-    %% Both the client and rabbitmq server were running on the same machine, a MacBookPro laptop.
-    %%
-    %% rabbitmq.config was created in its default location for OS X: /usr/local/etc/rabbitmq/rabbitmq.config.
-    %%
-    %% The contents of the example rabbitmq.config are for demonstration purposes only. See https://www.rabbitmq.com/ssl.html for instructions about creating the test certificates and the contents of rabbitmq.config.
-
-
-    [
-      {rabbit,
-        [
-          {ssl_listeners, [{"127.0.0.1", 5671}]},
-
-          %% Configuring SSL.
-          %% See http://www.rabbitmq.com/ssl.html for full documentation.
-          %%
-          {ssl_options, [{cacertfile,           "/Users/me/tls-gen/basic/testca/cacert.pem"},
-                         {certfile,             "/Users/me/tls-gen/basic/server/cert.pem"},
-                         {keyfile,              "/Users/me/tls-gen/basic/server/key.pem"},
-                         {verify,               verify_peer},
-                         {fail_if_no_peer_cert, true}]}
-        ]
-      }
-    ].
+    # Enable A.M.Q.P.S.
+    listeners.ssl.default = 5671
+    ssl_options.cacertfile = PIKA_DIR/testdata/certs/ca_certificate.pem
+    ssl_options.certfile = PIKA_DIR/testdata/certs/server_certificate.pem
+    ssl_options.keyfile = PIKA_DIR/testdata/certs/server_key.pem
+    ssl_options.verify = verify_peer
+    ssl_options.fail_if_no_peer_cert = true
+
+    # Enable H.T.T.P.S.
```

Comment:
Can we change this to just "HTTPS" (this is not a music album cover ;)) and add a comment that says "See http://www.rabbitmq.com/ssl.html to learn more"? The rest looks reasonable to me.

Reply:
From [Wikipedia](https://en.wikipedia.org/wiki/Acronym#Punctuation):

> In English, abbreviations have traditionally been written with a full stop/period/point in place of the deleted part to show the ellipsis of letters – although the colon and apostrophe have also had this role – and with a space after full stops (e.g. "A. D."). In the case of most acronyms, each letter is an abbreviation of a separate word and, in theory, should get its own termination mark. Such punctuation is diminishing with the belief that the presence of all-capital letters is sufficient to indicate that the word is an abbreviation.
> Some influential style guides, such as that of the BBC, no longer require punctuation to show ellipsis; some even proscribe it. Larry Trask, American author of The Penguin Guide to Punctuation, states categorically that, in British English, "this tiresome and unnecessary practice is now obsolete".
> Nevertheless, some influential style guides, many of them American, still require periods in certain instances. For example, The New York Times' guide recommends following each segment with a period when the letters are pronounced individually, as in K.G.B., but not when pronounced as a word, as in NATO. The logic of this style is that the pronunciation is reflected graphically by the punctuation scheme.

Since periods are disappearing, I’ve just removed them (but note that they’re *not* wrong). Done.

Reply:
> add a comment that says "See http://www.rabbitmq.com/ssl.html to learn more"?

It’s actually already there at the beginning of the example so it would be a repetition:

> See https://www.rabbitmq.com/ssl.html for certificate generation and RabbitMQ TLS configuration.